src/Moryx.Products.UI.Interaction/Recipes/Details/RecipeDetailsViewModelBase.cs

```diff
@@ -81,16 +81,19 @@ public async Task Load(RecipeViewModel recipeVm, IReadOnlyCollection<WorkplanVie
             await LoadTypeAndWorkplans(recipeVm.Model, workplans);
 
             WorkplanViewModel wp = null;
-            if (recipeVm.Model.WorkplanId != 0)
-                wp = workplans.FirstOrDefault(w => w.Id == recipeVm.Model.WorkplanId);
+            if (recipeVm.Model.WorkplanModel.Id != 0)
+                wp = Workplans.FirstOrDefault(w => w.Id == recipeVm.Model.WorkplanModel.Id && w.Version == recipeVm.Model.WorkplanModel.Version);
```

Comment:
Comparing versions is not necessary, as earch version of a workplan has a new id.